### DISCUSSION TITLE
Convert the kinematic orbits from CRF to along-track, cross-track and radial-track

### DISCUSSION THREAD

**Discussion by ParisaShafiei** [groops-devs/groops#121] [Q&A] [answered]
Dear developers, 

I have the kinematic orbits of GRACE-FO in the Celestial Reference Frame (CRF) and I would like to convert them to the along-track, cross-track, and radial-track directions. I went through the user manual and I could not find any programs to convert the orbits from CRF to along-track, cross-track and radial-track. 

Could you please tell me which program can be used for this purpose? 

Thank you in advance for your time and help.

With Kind Regards, 
Parisa

**Comment by tmayerguerr** [comment, marked as answer]
Dear Parisa,

You can use [SimulateStarCamera](https://groops-devs.github.io/groops/html/SimulateStarCamera.html) to generate rotations from orbit system to CRF and apply the rotations with [InstrumentRotate](https://groops-devs.github.io/groops/html/InstrumentRotate.html) to your orbit data (with inverseRotate).

regards
Torsten

**Comment by ParisaShafiei** [comment]
Dear Torsten, 

Thank you so much for your help. 

With Kind Regards, 
Parisa

**Comment by ParisaShafiei** [comment]
Dear Torsten, 

Regarding the last question I ask about converting satellite orbits form CRF to along track, cross track and radial, II have used the program called "InsrumentRotate" with inverse rotate to go from CRF to local orbit frame. My purpose is to validate the results I have gotten in GROOPS for the mission GRACE FO with those of AIUB. 

First: I used the program "Bernese2kinematicorbit" to read the AIUB orbits in TRF and to compare them with GROOPS results. At this step, the residuals are small enough in CRF.

Second: To have a better understanding, I converted those orbits to local orbit frame for both GROOPS and AIUB results. To my surprise, even though the star cameras computed for both GROOPS and AIUB results, the residuals of the final orbits in along track, cross track and radial are not small especially in the along track values which are so big. 

I have tried it several times and even each time I have the same problem. Even I tried with the final processed TUG orbits for this mission and still I have the same problem. (Please check the attached file)

[TUG_vs_AIUB_residuals_2020-01-01.txt](https://github.com/groops-devs/groops/files/9952147/TUG_vs_AIUB_residuals_2020-01-01.txt)

Could you please tell me if there is something that I should consider to solve this issue?

Thank you so much in advance.

With Kind Regards, 
Parisa

**Comment by tmayerguerr** [comment]
Dear Parisa,

I can only guess.

Here the steps for clarification
1. [SimulateStarCamera](https://groops-devs.github.io/groops/html/SimulateStarCamera.html) using one of the orbits (not the residuals)
2. Compute the difference between the two orbits
3. Apply the inverse rotation with [InstrumentRotate](https://groops-devs.github.io/groops/html/InstrumentRotate.html) to the difference

The length of the residual vector sqrt(data0^2+data1^2+data2^2) should be the same after step 2 and after step3.

regards
Torsten

**Reply by ParisaShafiei** [reply]
Dear Torsten, 

First of all, thanks for your feedback. 

I tried the 3 steps that you have mentioned to figure out if the length of the residual vectors after step 2 and step 3 are the same or not by comparing my results with the final TUG estimated orbits.  

At first, I added velocity and acceleration to the orbits in CRF and then simulated the star camera. Following step 2, I got the residuals for one day (please see the attached file) and then, when I wanted to apply the inverse rotation to this file, I got the following error message: 

**rotation for MISCVALUES(3) not implemented  continue...**

I still have doubts why when I calculate the orbit residuals in CRF, they are much smaller than those of in LOF by using the estimated orbits in this frame. 

**My orbits in CRF and then converted to LOF:**

[gracefo1_orbit_in_LOF_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069761/gracefo1_orbit_in_LOF_2020-01-01.txt)
[gracefo1_starcamera_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069763/gracefo1_starcamera_2020-01-01.txt)

**TUG downloaded orbits in CRF and then converted to LOF:**

[GRACEFO-1_orbit_in_LOF_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069765/GRACEFO-1_orbit_in_LOF_2020-01-01.txt)

[GRACEFO-1_starcamera_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069772/GRACEFO-1_starcamera_2020-01-01.txt)

**Residuals in CRF:**
[Res_Orbit_in_CRF_UNI.LU_vs_TUG_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069784/Res_Orbit_in_CRF_UNI.LU_vs_TUG_2020-01-01.txt)

**Residuals in LOF:**
[Res_Orbit_in_LOF_UNI.LU_vs_TUG_2020-01-01.txt](https://github.com/groops-devs/groops/files/10069787/Res_Orbit_in_LOF_UNI.LU_vs_TUG_2020-01-01.txt)

I would be pleased if you could take a look at them, I am really wondering where the problem originates from. 

Thank you in advance for your time and help. 

With kind regards
Parisa

**Comment by ParisaShafiei** [comment]
Dear Torsten, 

There is an update:

I would like to inform you that I found my mistake in converting the orbit differences from CRF to LOF. By using the "vector3D"option in "InstrumentArcCalculate", I got the orbits differences in along, cross, and radial. 

I hope this discussion will be useful for the other users to avoid my mistake. 

With Kind Regards, 
Parisa
